src/Monitor/Monitor/Alerts/NewAzureRmMetricAlertRuleV2CriteriaCommand.cs

```diff
@@ -61,78 +74,85 @@ public class NewAzureRmMetricAlertRuleV2CriteriaCommand : MonitorCmdletBase
         [Parameter(Mandatory = true, HelpMessage = "The rule condition operator")]
-        [PSArgumentCompleter("GreaterThan", "GreaterThanOrEqual", "LessThan", "LessThanOrEqual")]
+        [PSArgumentCompleterWithScript(OperatorCompleterScript)]
         public String Operator { get; set; }
 
         /// <summary>
         /// Gets or sets the rule threshold
         /// </summary>
         [Parameter(ParameterSetName = StaticThresholdParameterSet, Mandatory = true,  HelpMessage = "The threshold for rule condition")]
         public double Threshold { get; set; }
-
-        /// <summary>
-        /// Gets or sets the rule DynamicThreshold
-        /// </summary>
-        [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = true, HelpMessage = "The Dynamic Threshold for rule condition")]
-        public string DynamicThreshold { get; set; }
-
+        
         /// <summary>
         /// Gets or sets the rule sensitivity
         /// </summary>
         [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = false, HelpMessage = "The sensitivity for rule condition")]
         [PSArgumentCompleter("Low", "Medium", "High")]
-        public String Sensitivity { get; set; }
+        public String ThresholdSensitivity { get; set; } = "Medium";
 
         /// <summary>
-        /// Gets or sets the rule FailingPeriod
+        /// Gets or sets the rule Number of violated points
         /// </summary>
-        [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = false, HelpMessage = "The Failing Period for rule condition")]
-        public int FailingPeriod { get; set; }
+        [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = false, HelpMessage = "The minimum number of violations required within the selected lookback time window required to raise an alert")]
+        public int NumberOfViolations { get; set; } = 4;
 
         /// <summary>
         /// Gets or sets the rule TotalPeriod
         /// </summary>
-        [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = false,  HelpMessage = "The Total Period for rule condition")]
-        public int TotalPeriod { get; set; }
+        [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = false, HelpMessage = "The Total number of examined points")]
+        public int NumberOfExaminedAggregatedPoints { get; set; } = 4;
 
         /// <summary>
         /// Gets or set IgnoreDataBefore  parameter
         /// </summary>
-        [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = false, HelpMessage = "The IgnoreDataBefore parameter")]
-        public DateTime IgnoreDataBefore { get; set; }
+        [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = false, HelpMessage = "The date from which to start learning the metric historical data and calculate the dynamic thresholds")]
+        public DateTime? IgnoreDataBefore { get; set; }
```

Comment:
This is not technically a breaking change, but it violates the PowerShell convention – value type parameters should not be nullable, as the code can check whether the customer passed a value in using the `BoundParameters` dictionary (and we supply an extension method `IsParameterBound` that devs should use for this purpose.